### DISCUSSION TITLE
Adding Sprig for mutating outputs -> vars

### DISCUSSION THREAD

**Discussion by coryodaniel** [getporter/porter#1364] [Ideas]
**Is your feature request related to a problem? Please describe.**

Sometimes an output needs to be manipulated for an input to another mixin invocation.

Say a bundle has two mixin invocations. 

The first one has an output that’s a list of IPs. 

The second one needs a single IP address as an input.

Using sprig we could split the IP list on comma and take the first IP. 

{{ bundle.outputs.ip_list | splitList “,” | first }}

**Describe the solution you'd like**
Including the [sprig](http://masterminds.github.io/sprig/) library to provide more template functions.

**Comment by carolynvs** [comment]
Thanks for the pointer to sprig! We most definitely want to have stronger templating features so that transforming outputs is easier.

**Comment by carolynvs** [comment]
We are just embarking on adding a [proposal process](https://github.com/getporter/proposals/blob/main/pep/001-initial-pep-process.md) to Porter. The first step is discussing the idea in our forums (I've moved the issue into the forum). 

There is an [existing issue](https://github.com/getporter/porter/issues/316) that discusses the idea of adding functions to our templates, but there has been no traction.

For context, we currently use [cbroglie/mustache](https://github.com/cbroglie/mustache) for templating. So a proposal would need to take that into consideration, since it looks like sprig uses go templates. We'd want to understand if it would be a breaking change, or if another library would work well. One concern that the mentioned issue gets into is how to use templating without creating invalid yaml, and still allow the expression of non-string values. So that would be useful to keep in mind as well.

I'm not asking that you write up a PEP, though if you'd like to that would be incredibly helpful. 😉  Just leaving info here for whoever would like to tackle this.
